hoprd/rest-api/src/messages.rs

```diff
@@ -127,24 +126,60 @@ pub(super) async fn send_message(
             .into_response()
     })?;
 
+    let destination = args.clone().destination.fulfill(hopr.peer_resolver()).await;
+
+    let peer_id = match destination {
+        Ok(destination) => match destination.peer_id {
+            Some(peer_id) => peer_id,
+            None => return Err((StatusCode::NOT_FOUND, ApiErrorStatus::InvalidInput).into_response()),
+        },
+        Err(e) => return Err(e.into_response()),
+    };
+
     // Use the message encoder, if any
     let msg_body = state
         .msg_encoder
         .as_ref()
         .map(|enc| enc(&args.body))
         .unwrap_or_else(|| args.body.into_boxed_slice());
 
-    let options = if let Some(intermediate_path) = args.path {
-        RoutingOptions::IntermediatePath(intermediate_path.try_into().map_err(|_| {
-            (
-                StatusCode::UNPROCESSABLE_ENTITY,
-                ApiErrorStatus::UnknownFailure(format!(
-                    "Intermediate path cannot be longer than {}",
-                    RoutingOptions::MAX_INTERMEDIATE_HOPS
-                )),
-            )
-                .into_response()
-        })?)
+    let path = args.path;
+
+    // TODO (jean): find a nice way to handle PeeroOrAddress fulfilling
```

Comment:
So if I understand correctly - if now I supply the path made of chain addresses, the code will simply  panic due to the `unwrap` ?

This is not good. 

You need to either `fulfill` all the `PeerOrAddress`  instances in `args.path` or revert that parameter to be `Vec<PeerId>`, which however, defeats the purpose of this PR.

Reply:
I was stuck at handling the fulfilling of all the identifer provided in the `path`, mainly handling an error if any occurs.
My attempt is commented right bellow uner the `TODO (jean)`